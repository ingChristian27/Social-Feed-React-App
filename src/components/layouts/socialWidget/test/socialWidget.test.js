import React from "react";
import { create, act } from "react-test-renderer";
import * as reactRedux from "react-redux";
import SocialWidget from "../index";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk]


const data = [
    {
        created_at: "Fri Dec 29 19:15:04 +0000 2017",
        id: 946821889648980000,
        id_str: "946821889648979968",
        name: "Lori King",
        text: "The battle is life or death, but that doesn’t mean love plays no part. ↵#shifters #menage #paranormal #ebook… https://t.co/61xUthkXFN",
    },
];

describe("Snapshow Widget", () => {
    const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
    const initialState = { data };
    const mockStore = configureStore(middlewares)
    beforeEach(() => {
        useSelectorMock.mockReturnValue(data);
    });

    it("Snapshow Widget", async () => {
        let component;
        await act(async () => {
            const store = mockStore(initialState);

            component = create(
                <Provider store={store}>
                    <SocialWidget></SocialWidget>
                </Provider>
            );
            
        });

        await act(async () => {
            const store = mockStore(initialState);
            component.update(
                <Provider store={store}>
                    <SocialWidget></SocialWidget>
                </Provider>
            );
            
        });

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
