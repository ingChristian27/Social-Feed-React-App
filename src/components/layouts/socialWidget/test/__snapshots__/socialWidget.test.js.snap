// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshow Widget Snapshow Widget 1`] = `
<div
  className="sc-dkPtyc iGpeZR"
>
  <section
    className="sc-bdvvaa xXxIo"
  >
    <div
      className="sc-gsDJrp fCYfGI"
    >
      <div
        className="sc-gsDJrp fmguIX"
      >
        <div
          className="sc-gKckTs fDZbFu"
        >
          <p
            className="sc-hKwCoD hJtrdA"
            color="primary"
            fontStyle="ligth"
          >
            The battle is life or death, but that doesn’t mean love plays no part. ↵#shifters #menage #paranormal #ebook… https://t.co/61xUthkXFN
          </p>
          <div
            className="sc-furvIG eZcjDD"
          >
            - 
            Lori King
          </div>
        </div>
        <div
          className="sc-iCfLBT cSyTin"
        >
          Posted: 
          Fri Dec 29 19:15:04 +0000 2017
        </div>
      </div>
    </div>
  </section>
</div>
`;
